channelserver/src/main.rs

```diff
@@ -94,6 +98,7 @@ fn channel_route(req: &HttpRequest<session::WsChannelSessionState>) -> Result<Ht
         "remote_ip" => &meta_info.remote
     );
 
+    // Cannot apply headers here.
```

Comment:
Looking into the code, I don't think I can, or at least, not without some weird surgery. The problem is that ws::start generates the HttpResponse and then posts it independently. So I can't append the headers before it's sent out. 

I didn't think about the DefaultHeader middleware, and yeah, these are fairly universal. I'll do the refactor later since I want to address a few things about that. Thanks!